### PR TITLE
Added drag/drop file support removed FF22 paste

### PR DESCRIPTION
**Pull request by Buildstarted**
- Removed duplicate functionality from HomeModule.cs
  to ensure only one upload method
- Added drag/drop functionality to the ui
- Removed Firefox paste handling code
- Cleaned up file upload js to remove code duplication

Drag/Drop UI

### CODE REVIEW

File: JabbR/Nancy/HomeModule.cs [hunk 1]
```diff
@@ -91,16 +67,21 @@ public class HomeModule : JabbRModule
...
-                    var info = Regex.Match(file, @"data:image/(?<type>.+?);base64,(?<data>.+)");
+                    var info = Regex.Match(file, @"data:(?:(?<unkown>.+?)/(?<type>.+?))?;base64,(?<data>.+)");
 
-                    var binData = Convert.FromBase64String(info.Groups["data"].Value);
+                    binData = Convert.FromBase64String(info.Groups["data"].Value);
                     contentType = info.Groups["type"].Value;
 
-                    fileName = fileName + "." + contentType.Substring(contentType.IndexOf("/") + 1);
-
+                    if (string.IsNullOrWhiteSpace(contentType))
```
Comment (davidfowl) [inline JabbR/Nancy/HomeModule.cs]: Upper case S

File: JabbR/Chat.ui.js [hunk 2]
```diff
@@ -731,27 +731,45 @@
         $previewUpload.modal();
     }
 
-    function initializeUploadPreview(e, file, type) {
```
Comment (davidfowl) [inline JabbR/Chat.ui.js:734]: I think we should split this file uploader into another file.

Comment (Buildstarted) [inline JabbR/Chat.ui.js:734]: Would it be fine if I put the paste code in the same file? (since it's all related)

Comment (davidfowl) [inline JabbR/Chat.ui.js:734]: yep

File: JabbR/Chat.ui.fileUpload.js [hunk 3]
```diff
@@ -0,0 +1,300 @@
+﻿(function ($, ui) {
+
+    console.log("initializing file upload");
```
Comment (davidfowl) [inline JabbR/Chat.ui.fileUpload.js]: Remove this

**Comment by davidfowl** [comment]
@Buildstarted can you rebase again?

**Pull request merged**